src/datastore/postgres-store.ts

```diff
@@ -820,9 +820,29 @@ export class PgDataStore
       UPDATE names
       SET canonical = $2
       WHERE index_block_hash = $1 AND canonical != $2
+      RETURNING name
       `,
       [indexBlockHash, canonical]
     );
+    nameResult.rows.forEach(async name => {
```

Comment:
`Array.forEach` is going to execute all these promises in parallel, then exit the function and leave these as dangling promises, with the `client` object instance being disposed off while in use. 
These need switched to a regular `for` loop with `await`s within the loop body.